File: examples/Python.ExampleApplication/services/migration_content_loader/empty_migration_content_loader.py
from typing import TypeVar
from tableau_migration import (
    empty_pager,
    MigrationContentLoaderBase
)

TContent = TypeVar("T")

class EmptyMigrationContentLoader(MigrationContentLoaderBase[TContent]):
    def get_migration_content_pager(self, page_size: int):
        return empty_pager(TContent)

File: examples/Csharp.ExampleApplication/Services/MigrationContentLoader/EmptyMigrationContentLoader.cs
using Tableau.Migration;
using Tableau.Migration.Engine.Endpoints;
using Tableau.Migration.Paging;

namespace Csharp.ExampleApplication.Services.MigrationContentLoader
{
    #region class
    public class EmptyMigrationContentLoader<TContent> : IMigrationContentLoader<TContent>
        where TContent : IContentReference
    {
        public IPager<TContent> GetMigrationContentPager(int pageSize)
            => new MemoryPager<TContent>([], pageSize);
    }
    #endregion
}

